File: packages/python/src/alumnium/tools/print_to_pdf_tool.py
from pydantic import Field

from alumnium.drivers.base_driver import BaseDriver

from .base_tool import BaseTool


class PrintToPdfTool(BaseTool):
    """Print the current page to a PDF file."""

    filepath: str = Field(description="Path to save the PDF file to")

    def invoke(self, driver: BaseDriver):
        driver.print_to_pdf(self.filepath)

File: packages/typescript/src/tools/PrintToPdfTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";
import { field, type FieldMetadata } from "./Field.ts";

export class PrintToPdfTool extends BaseTool {
  static description = "Print the current page to a PDF file.";
  static fields: FieldMetadata[] = [
    field({
      name: "filepath",
      type: "string",
      description: "Path to save the PDF file to",
    }),
  ];

  filepath: string;

  constructor(args: { filepath: string }) {
    super();
    this.filepath = args.filepath;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.printToPdf(this.filepath);
  }
}
